content/posts/annoucing-slight-v0_3.md

```diff
@@ -0,0 +1,226 @@
+
+### HTTP Client
+
+Previously, `Slight` only supports the ability to serve HTTP requests. You can write applications in Rust or C to handle incoming HTTP requests and compile them to Wasm modules. And now, `Slight` supports the ability to make a outbound HTTP request to a url. Take the following Rust program as an example:
+
+```rust
+wit_bindgen_rust::import("wit/http-client.wit");
+
+#[register_handler]
+fn handle_hello(_req: Request) -> Result<Response, HttpError> {
+    let req = crate::http_client::Request {
+        method: crate::http_client::Method::Get,
+        uri: "https://some-random-api.ml/facts/dog",
+        headers: &[],
+        body: None,
+        params: &[],
+    };
+    let res = crate::http_client::request(req).unwrap();
+    println!("{:?}", res);
+
+    let res = Response {
+        status: res.status,
+        headers: res.headers,
+        body: res.body,
+    };
+    Ok(res)
+}
+```
+
+You can use `wit-bindgen` tool to generate guest binding from `http-client.wit` WIT file. 
+
+> 💡 The WIT file can be downloaded using `slight add http-client@v0.3.1`
+
+The `handle_hello` function is responsible for handling incoming HTTP requests. It achieves this by creating a `Request` object from the generated `http_client` module, which is then sent using the `request` function. The `request` function makes an HTTP request to the URI defined in the `Request` struct.
+
+It is worth noting that the `request` function is also generated. Its purpose is to handle the actual sending of the `Request` object. Once the `request` function has completed this task, it returns a `Response` object, which serves as the return value for the `handle_hello` function.
+
+Developers can easily interface with the HTTP request processing mechanism by using the `handle_hello` function. They can be confident that their HTTP requests will be handled appropriately. This function is a crucial part of any HTTP-based application.
+
+Lastly, to enable both `HTTP` and `HTTP-client` capabilities, we need to update the configuration file. Below is an example slightfile.toml file.
+
+```rust
+specversion = "0.2"
+
+[[capability]]
+resource = "http"
+name = "my-rest-api"
+    # This capability does not require any configs
+
+[[capability]]
+resource = "http-client"
+name = "something"
+```
+
+### Exchange Messaging
+
+In this new release, we have combined the `message-queue` and `pub/sub` capabilities into one called `messaging`. The idea behind this move is that we think there are many overlaps and ambiguities to both capabilities and it made a difficult time for developers to choose one over another. If you want to know more about the design of the new messaging capability, please read this [proposal](https://github.com/deislabs/spiderlightning/blob/main/proposals/abandoned/pubsub-mq-interfaces-proposal.md) document that Dan drafted. 
```

Comment:
Oh nice! I like how you noticed this was confusing to users and improved the abstraction to make it align better with how people think about it. 👍